v1/lists.yaml

```diff
@@ -0,0 +1,1011 @@
+          in: query
+          required: true
+          schema: *csrf_token
+      responses:
+        '200':
+          description: Success.
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/problem'
+      x-route-filters:
+        - path: ./lib/mediawiki_auth_filter.js
+      x-request-handler:
+        - forward_to_mw:
+            request:
+              uri: /{domain}/sys/action/rawquery
+              body:
+                action: readinglists
+                command: teardown
+                token: '{{request.query.csrf_token}}'
+            return:
+              status: 200
+              headers:
+                content-type: application/json; charset=utf-8
+                cache-control: "no-cache"
+      x-monitor: false
+  /lists/:
+    get:
+      tags:
+        - Reading lists
+      summary: Get all lists of the current user.
+      description: |
+        Returns metadata describing the lists of the current user. Might be truncated and include
+        a continuation token.
+
+        Request must be authenticated with a MediaWiki session cookie.
+
+        Stability: [unstable](https://www.mediawiki.org/wiki/API_versioning#Unstable)
+      parameters:
+        - name: next
+          in: query
+          description: Continuation parameter from previous request
+          type: string
+          required: false
+      produces:
+        - application/json; charset=utf-8; profile="https://www.mediawiki.org/wiki/Specs/Lists/0.1"
+      responses:
+        '200':
+          description: An array of list metadata.
+          schema:
+            type: object
+            properties:
+              lists:
+                type: array
+                items:
+                  $ref: '#/definitions/list_read'
+              next:
+                type: string
+                description: Continuation token.
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/problem'
+      x-route-filters:
+        - path: ./lib/mediawiki_auth_filter.js
+      x-request-handler:
+        - forward_to_mw:
+            request:
+              uri: /{domain}/sys/action/rawquery
+              body:
+                action: query
+                meta: readinglists
+                rllimit: max
+                continue: '{{unflattenContinuation(request.query.next, 1).continue}}'
+                rlcontinue: '{{unflattenContinuation(request.query.next, 1).rlcontinue}}'
+            return:
+              status: 200
+              headers:
+                content-type: application/json; charset=utf-8; profile="https://www.mediawiki.org/wiki/Specs/Lists/0.1"
+                cache-control: "no-cache"
+              body:
+                lists: '{{forward_to_mw.body.query.readinglists}}'
+                next: '{{flattenContinuation(forward_to_mw.body.continue)}}'
+      x-monitor: true
+      x-amples:
+        - title: Get all my lists
```

Comment:
How will this get all the lists for a user if there is no token to identify them?

Reply:
Good question. I was thinking of adding the session cookie to the Swagger spec as a security scheme object with a vendor extension, and then making service-checker handle it. Do you think that makes sense? I am writing some mocha tests that have a flag for using nock (much like `tests/pagecontent/save_api.js`) so that could also be used to do smoke testing (including write endpoints) - I guess that would make `x-monitor` pointless?

In any case I'll remove it from this commit. It can be readded later of there is a sensible way of using it.